File: lightscript/lightscript2/stdlib.js
var GLOBAL = (function () {
    return this;
})();
if (! GLOBAL . readline) {
    readline = (function () {
        importPackage(java . io);
        importPackage(java . lang);
        stream = BufferedReader(InputStreamReader(System["in"]));
        return function () {
            var line = stream . readLine();
            if (line !== null) {
                return "" + line;
            } else {
                return "";
            };
        };
    })();
};
var getch = (function () {
    var line = "";
    var pos;
    var newl = 0;
    return function () {
        pos = pos + 1;
        if (line[pos] !== undefined) {
            newl = 0;
            return line[pos];
        } else {
            pos = -1;
            line = readline();
            newl = newl + 1;
            if (newl > 10) {
                return undefined;
            } else {
                return "\n";
            };
        };
    };
})();
// Function that prettyprints a list of lists
function tail(list, n) {
    n = n || 1;
    return list.slice(n);
}
function slice(list, a, b) {
    return list.slice(a, b);
}
function put(obj, key, val) {
    obj[key] = val;
}
function map(fn, list) {
    return list.map(fn);
}
function array_join(list, sep) {
    return list.join(sep);
}
function array_push(list, elem) {
    return list.push(elem);
}
function string_contains(str, c) {
    return str.indexOf(c) != -1;
}
function array() {
    return Array.prototype.slice.call(arguments);
}
function strjoin() {
    return Array.prototype.slice.call(arguments).join("");
}
function len(obj) {
    return obj.length;
}
function get_type(obj) {
    return typeof(obj);
}
function is_array(obj) {
    return obj && obj.constructor === Array
}
function is_string(obj) {
    return typeof (obj) === "string"
}
function idx(obj, id) {
    return obj[id];
}
function get(obj, id, default_value) {
    return obj[id] || default_value;
}
function cons(obj, arr) {
    var result = arr.slice(0);
    result.unshift(obj);
    return result;
}
function assert(ok) {
    if(!ok) throw "ASSERT ERROR";
}
STRING_TYPE = get_type("")

File: lightscript/lightscript2/stdlib.py
import sys
undefined = None
true = True
false = False
def keys(obj):
    return obj.keys()

def contains(str, c):
    return c in str

def getch():
    c = sys.stdin.read(1)
    if len(c) == 0:
        return undefined
    return c

def tail(list, n = None):
    if n == None:
        n = 1
    return list[n:]

def fold(fn, list, acc):
    for i in range(len(list)):
        acc = fn(list[i], acc)
    return acc;

def array_join(list, sep):
    return sep.join(list)

def array_push(list, elem):
    list.append(elem)

def push(list, elem):
    list.append(elem)
    return list

def string_contains(str, c):
    return c in str

def strjoin(*args):
    return "".join(args)

def get_type(obj):
    return type(obj)

def is_array(obj):
    return type(obj) == type([])

def is_string(obj):
    return type(obj) == type("")

def get(obj, id, default_value):
    return obj.get(id, default_value)

def cons(obj, arr):
    result = [obj]
    result.extend(arr)
    return result

arguments = sys.argv[1:]
